### PR TITLE
charls: add version 2.2.0 with modernized

### PR DESCRIPTION
**Pull request by toge**
Specify library name and version:  **charls/2.2.0**

All paches in 2.1.0 are merged by 2.2.0.

---

- [x] I've read the [guidelines](https://github.com/conan-io/conan-center-index/blob/master/docs/how_to_add_packages.md) for contributing.
- [x] I've followed the [PEP8](https://www.python.org/dev/peps/pep-0008/) style guides for Python code in the recipes.
- [x] I've used the [latest](https://github.com/conan-io/conan/releases/latest) Conan client version.
- [x] I've tried at least one configuration locally with the
      [conan-center hook](https://github.com/conan-io/hooks.git) activated.

### CODE REVIEW

File: recipes/charls/all/test_package/CMakeLists.txt [hunk 1]
```diff
@@ -2,12 +2,14 @@ cmake_minimum_required(VERSION 3.1.3)
...
 
 include(${CMAKE_BINARY_DIR}/conanbuildinfo.cmake)
-conan_basic_setup()
+conan_basic_setup(TARGETS)
+
+find_package(charls CONFIG REQUIRED)
 
 add_executable(test_package_c test_package.c)
-target_link_libraries(test_package_c ${CONAN_LIBS})
+target_link_libraries(test_package_c charls::charls)
```
Comment (SpaceIm) [inline recipes/charls/all/test_package/CMakeLists.txt:10]: official target is not namespaced I think: https://github.com/team-charls/charls/blob/3dd2e03a83a4d69825c984f7a7e8e319ecae1cd2/src/CMakeLists.txt#L114

Comment (toge) [reply recipes/charls/all/test_package/CMakeLists.txt:10]: I have missed it.
I will fix later.

Comment (toge) [reply recipes/charls/all/test_package/CMakeLists.txt]: I will try to fix it.
Cloud you check my modification?
I'm afraid I don't follow your comment.

File: recipes/charls/all/conanfile.py [hunk 2]
```diff
@@ -1,17 +1,19 @@
 import os
+import textwrap
 
 from conans import ConanFile, CMake, tools
 from conans.errors import ConanInvalidConfiguration
 
+required_conan_version = ">=1.33.0"
```
Comment (SpaceIm) [inline recipes/charls/all/conanfile.py:7]:
```suggestion
required_conan_version = ">=1.36.0"

```

File: recipes/charls/all/conanfile.py [hunk 3]
```diff
@@ -59,12 +83,34 @@ def _configure_cmake(self):
...
+        tools.rmdir(os.path.join(self.package_folder, "lib", "pkgconfig"))
+        self._create_cmake_module_alias_targets(
+            os.path.join(self.package_folder, self._module_file_rel_path),
+            {"charls": "charls::charls"}
+        )
 
     def package_info(self):
+        self.cpp_info.builddirs.append(self._module_subfolder)
+        self.cpp_info.build_modules["cmake_find_package"] = [self._module_file_rel_path]
+        self.cpp_info.build_modules["cmake_find_package_multi"] = [self._module_file_rel_path]
```
Comment (SpaceIm) [inline recipes/charls/all/conanfile.py:112]:
```suggestion
        self.cpp_info.set_property("cmake_file_name", "charls")
        self.cpp_info.set_property("cmake_target_name", "charls")
        self.cpp_info.builddirs.append(self._module_subfolder)
        self.cpp_info.set_property("cmake_build_modules", [self._module_file_rel_path])
```

**Review by SpaceIm — CHANGES_REQUESTED** [review]
let's move forward and use `set_property()`

File: recipes/charls/all/conanfile.py (hunk 3 above)
Comment (toge) [reply recipes/charls/all/conanfile.py]: Thank a lot!
Now I will know cmake property.

**Review by SpaceIm — APPROVED** [review]

**Review by uilianries — APPROVED** [review]
LGTM

File: recipes/charls/all/conanfile.py [hunk 4]
```diff
@@ -41,12 +61,17 @@ def configure(self):
             raise ConanInvalidConfiguration("CharLS can't be compiled by {0} {1}".format(self.settings.compiler,
                                                                                          self.settings.compiler.version))
 
+        # name lookup issue for gcc == 5 in charls/2.2.0
+        if self.settings.compiler == "gcc" and tools.Version(self.settings.compiler.version) == "5" and tools.Version(self.version) == "2.2.0":
```
Comment (ericriff) [inline recipes/charls/all/conanfile.py:65]: Does this belong in `validate()`?

**Review by SSE4 — APPROVED** [review]

**Pull request merged**